### DISCUSSION TITLE
ファームウェアの書き込みにおいてキーボードが認識されない

### DISCUSSION THREAD

**Discussion by sasaki9998** [Yowkees/keyball#196] [Q&A] [answered]
タイトルの件につきまして、[「3-8．ProMicroの書き込みと動作確認」](https://github.com/Yowkees/keyball/blob/main/keyball44/doc/rev1/buildguide_jp.md#3-8promicro%E3%81%AE%E6%9B%B8%E3%81%8D%E8%BE%BC%E3%81%BF%E3%81%A8%E5%8B%95%E4%BD%9C%E7%A2%BA%E8%AA%8D)の時点では、左右どちらのキーボードも認識出来まして、テスト用ファームウェアを書き込む事が出来ました。

現在は「[１０正規ファームウェアの書き込み](https://github.com/Yowkees/keyball/blob/main/keyball44/doc/rev1/buildguide_jp.md#%EF%BC%91%EF%BC%90%E6%AD%A3%E8%A6%8F%E3%83%95%E3%82%A1%E3%83%BC%E3%83%A0%E3%82%A6%E3%82%A7%E3%82%A2%E3%81%AE%E6%9B%B8%E3%81%8D%E8%BE%BC%E3%81%BF)」を実施しており、ファームウェアの書き込みにおいて左右どちらのキーボードも認識されない事象が発生しています。

[こちらのページ](https://remap-keys.app/catalog/tAJ9Htme4oNabUkx4832/firmware)からFLASHの実行時に「remap-keys.appがシリアルポートへの接続を要求しています」のメニューが表示されますが、対象リストの中にkeyballがありません。

半田付けしたRESETスイッチを素早く2回押しましたが、状況は変わりませんでした。
解決方法についてご教示いただけますでしょうか。

**Comment by katsusuke** [comment, marked as answer]
> [こちらのページ](https://remap-keys.app/catalog/tAJ9Htme4oNabUkx4832/firmware)からFLASHの実行時に「remap-keys.appがシリアルポートへの接続を要求しています」のメニューが表示されますが、対象リストの中にkeyballがありません。

ファームウエアを書き込むときは、デバイスの名称は keyball にはならないようです。
私の環境ではArduino Leonardo (ポート番号) 又は Pro Micro (ポート番号)でした(Mac / Windows 両方とも同じでした)

<img width="997" alt="スクリーンショット 2023-02-08 17 30 16" src="https://user-images.githubusercontent.com/624736/217475995-ce633dd8-0abd-4157-9684-1ee3e82ddb33.png">

スクリーンショットはMacです

ちなみに私の環境でのFLASH の手順ですが、

1. PCとUSBを接続する
2. Remap のサイトで、FLASH ボタンを押す
3. Chrome のメニューからデバイスリストが出る(この時点では Arduino Leonardo は表示されない)
4. reset ボタンを2回押す
6. この状態で、デバイスリストに Arduino Leonardo が表示される
7. 素早くArduino を選択して、接続ボタンを押す(10秒くらいするとタイム・アウトして、デバイスが消えてしまうので素早く押す)
8. ファームが書き込まれる

という感じでした

私も初心者ですがお役に立てたら幸いです。

**Comment by sasaki9998** [comment]
@katsusuke san
ご教示いただきありがとうございます。
reset ボタンは、デバイスリストが表示されてから押されているのですね。
デバイスリストの表示前にreset ボタンを押しておりました。

Promicro、Resetボタン周りのはんだ付けをやり直し、
デバイスリストが表示されてからreset ボタンを押すことで無事keyballを認識いたしました。
ご助言いただき大変助かりました。

@Yowkees san
お騒がせいたしました。本件クローズで問題ございません。
ファームウェアの書き込みまで出来ましたので、大切に使わせていただきます。

**Comment by Yowkees** [comment]
お休みをいただいておりご返信が遅れて申し訳ありませんでした。
そうですね。FLASHを押した後にRESETするとデバイスが出てくると思います。
ProMicroの初回の書き込みは何もせずに書き込めるのですが、２回目からはリセット処理しないと書き込みできないのでつまずいたかもですね。

@katsusuke 様もご回答ありがとうございました！

以上、今後ともどうぞよろしくお願いいたします。
